### PR TITLE
FE: Profile payment #17

### PR DESCRIPTION
**Pull request by davidleecodes**
### What this PR does (required):
- is able to create stripe customer, store card, make a payment
- on visiting the profile payment, api call is made to get the secret
- if there is a payment object for the user the secret is returned
- if not payment object is created and then the secret is returned 

### Screenshots / Videos (required):
Screenshot 2021-06-30 163328

### Any information needed to test this feature (required):
- visa test card 4111111111111111

### Any issues with the current functionality (optional):
- List any problems where you were not able to complete all tasks on ticket
- List any problems this PR may cause for the project or other active PRs
- Post a screenshot/video of the problem, along with detailed console outputs where applicable

### CODE REVIEW

File: client/src/App.tsx [hunk 1]
```diff
@@ -18,47 +18,55 @@ import Messages from './pages/Messages/Messages';
 import ProtectedRoute from './components/ProtectedRoute';
 import Unauthorized from './pages/Unauthorize/Unauthorized';
 import SitterOwner from './pages/Sitters/SitterOwner';
+import { Elements } from '@stripe/react-stripe-js';
+import { loadStripe } from '@stripe/stripe-js';
 
 import './App.css';
 
+const stripePromise = loadStripe(
+  'pk_test_51J5B2eKVTqeqLi3vW1fqFgTEllljxwuWTGNwdOHOfzFZfSJeGlYisgmxohIEQLFUuNsefjYWrEtYr6errTkZfbq200PVHMyOyJ',
```
Comment (rajivtitus) [inline client/src/App.tsx:27]: Please store API key as an environment variable before merging.

File: server/stripe/index.js [hunk 2]
```diff
@@ -1,37 +1,73 @@
...
+const decodeToken = require("../utils/decodeToken");
+const Profile = require("../models/Profile");
+const User = require("../models/User");
+const Payment = require("../models/Payment");
+
+// @route GET /payment/secret
+// @desc returns secret, create payment if there isn't one
+const getPaymentSecret = async (req, res, next) => {
+  let decoded = decodeToken(req.cookies.token);
+  const userId = decoded.id;
```
Comment (rajivtitus) [inline server/stripe/index.js:12]: You can obtain the user's id from the request object, we don't need to decode the token.

File: client/src/App.tsx (hunk 1 above)
Comment (davidleecodes) [reply client/src/App.tsx:27]: I was going to store it as an environment variable before I created a PR, but I found this on the stripe site
https://stripe.com/docs/keys
```
There are also two types of API keys: publishable and secret.

Publishable API keys are meant solely to identify your account with Stripe, they aren’t secret. In other words, they can safely be published in places like your Stripe.js JavaScript code, or in an Android or iPhone app.
Secret API keys should be kept confidential and only stored on your own servers. Your account’s secret API key can perform any API request to Stripe without restriction.

File: server/stripe/index.js (hunk 2 above)
Comment (davidleecodes) [reply server/stripe/index.js:12]: I tried this in bookings controller, "POST" doesn't have `req.user.id`, "GET, PATCH, and DELETE" do

File: client/src/App.tsx (hunk 1 above)
Comment (rajivtitus) [reply client/src/App.tsx]: Thanks for sharing this resource!

**Pull request merged**